ms2deepscore/MetadataFeatureGenerator.py

```diff
@@ -1,34 +1,110 @@
-        if ionization_mode == "negative":
+
+class StandardScaler(MetadataFeatureGenerator):
+    def __init__(self, metadata_field: str, mean: float, std: float = None):
+        self.metadata_field = metadata_field
+        self.mean = mean
+        self.standard_deviation = std
+
+    def generate_features(self, metadata: Metadata):
+        feature = metadata.get(self.metadata_field, None)
+        assert self.metadata_field is not None, f"Metadata entry for {self.metadata_field} is missing."
+        assert isinstance(feature, (int, float))
+        if self.standard_deviation:
+            return (feature - self.mean) / self.standard_deviation
+        return feature - self.mean
+
+    @classmethod
+    def load_from_dict(cls, json_dict: dict):
+        """Create StandardScaler instance from json.
+        """
+        return cls(json_dict["metadata_field"],
+                   json_dict["mean"],
+                   json_dict["standard_deviation"],)
+
+
+class OneHotEncode(MetadataFeatureGenerator):
+    def __init__(self, metadata_field: str,
+                 entries_becoming_one):
+        self.metadata_field = metadata_field
+        self.entries_becoming_one = entries_becoming_one
+
+    def generate_features(self, metadata: Metadata):
+        feature = metadata.get(self.metadata_field, None)
+        assert self.metadata_field is not None, f"Metadata entry for {self.metadata_field} is missing."
+        if feature == self.entries_becoming_one:
+            return 1
+        return 0
+
+    @classmethod
+    def load_from_dict(cls, json_dict: dict):
+        """Create StandardScaler instance from json.
+        """
+        return cls(json_dict["metadata_field"],
+                   json_dict["entries_becoming_one"])
+
+
+class FeatureToBinary(MetadataFeatureGenerator):
+    def __init__(self, metadata_field: str,
+                 entries_becoming_one,
+                 entries_becoming_zero):
+        self.metadata_field = metadata_field
+        self.entries_becoming_one = entries_becoming_one
+        self.entries_becoming_zero = entries_becoming_zero
+
+    def generate_features(self, metadata: Metadata):
+        feature = metadata.get(self.metadata_field, None)
+        assert self.metadata_field is not None, f"Metadata entry for {self.metadata_field} is missing."
+        if feature == self.entries_becoming_one:
             return 1
-        assert False, "Ionization mode should be 'positive' or 'negative'"
+        if feature == self.entries_becoming_zero:
+            return 0
+        assert False, f"Feature should be {self.entries_becoming_one} or {self.entries_becoming_zero}, not {feature}"
+
+    @classmethod
+    def load_from_dict(cls, json_dict: dict):
+        """Create StandardScaler instance from json.
+        """
+        return cls(json_dict["metadata_field"],
+                   json_dict["entries_becoming_one"],
+                   json_dict["entries_becoming_zero"],)
+
+
+def load_from_json(json_dict):
```

Comment:
Confusing because it is named "from_json", asks for a "dict" but actually takes a list (if I got it right?).